src/core/config/border_agent.h

```diff
@@ -89,6 +89,38 @@
     OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_BORDER_AGENT_MESHCOP_SERVICE_ENABLE
+ *
+ * Define to 1 to enable Border Agent to manage registering/updating of the mDNS MeshCoP service(s) on the
+ * infrastructure link
+ *
+ * This includes the ephemeral key service when the `OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE` is enabled.
+ */
+#ifndef OPENTHREAD_CONFIG_BORDER_AGENT_MESHCOP_SERVICE_ENABLE
+#define OPENTHREAD_CONFIG_BORDER_AGENT_MESHCOP_SERVICE_ENABLE \
+    (OPENTHREAD_CONFIG_PLATFORM_DNSSD_ENABLE || OPENTHREAD_CONFIG_MULTICAST_DNS_ENABLE)
+#endif
+
+/**
+ * @def OPENTHREAD_CONFIG_BORDER_AGENT_MESHCOP_SERVICE_BASE_NAME
+ *
+ * Specifies the base name to construct the service instance name used when advertising the mDNS `_meshcop._udp`
+ * service by the Border Agent.
+ *
+ * Applicable when the `OPENTHREAD_CONFIG_BORDER_AGENT_MESHCOP_SERVICE_ENABLE` feature is enabled.
+ *
+ * The name can also be configured using the `otBorderAgentSetMeshCoPServiceBaseName()` API at run-time.
+ *
+ * Per the Thread specification, the service instance should be a user-friendly name identifying the device model or
+ * product. A recommended format is "<VendorName> <ProductName>".
+ *
+ * The name MUST have a length less than or equal to `OT_BORDER_AGENT_MESHCOP_SERVICE_BASE_NAME_MAX_LENGTH` (47 chars).
+ */
+#ifndef OPENTHREAD_CONFIG_BORDER_AGENT_MESHCOP_SERVICE_BASE_NAME
+#define OPENTHREAD_CONFIG_BORDER_AGENT_MESHCOP_SERVICE_BASE_NAME "otBorderAgent"
```

Comment:
Should we be closer to the recommended format? Something like "OpenThread BorderRouter"?

Also, since the instance name is typically more user visible, would "BorderRouter" make more sense than "BorderAgent"?

Reply:
I renamed it in the new pushed commit.

Ideally, vendors should choose a better name so this default would never be used. Maybe we should use `"Lazy Vendor Stupid Product"` and see how many of those appear in the future! 😄 

Regarding "border router" vs. "border agent," the service is really indicating that the device is acting as a BA and not necessarily a BR, but I guess we can say that today all BAs are indeed BRs as well.

Reply:
We used `"OpenThread BR (unspecified vendor)"` as default to make the name somewhat reasonable if the default is used, yet still highlight that the vendor did not properly set it.